base/multidimensional.jl

```diff
@@ -207,11 +207,11 @@ function _unsafe_getindex(::LinearIndexing, src::AbstractArray, I::AbstractArray
     D = eachindex(dest)
     Ds = start(D)
     s = 0
-    for b in eachindex(I)
+    @inbounds for b in eachindex(I)
```

Comment:
It's (currently) functionally the same, no?  As I understand it, we no longer propagate the inbounds state through the next calls, so those aren't affected by the `@inbounds` annotation.  Now, that may be something we decide to do later, so I can restrict this if you'd prefer… but I think that decision will have to consider blocks like this no matter what.